src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerEthSpec.scala

```diff
@@ -110,6 +106,48 @@ class JsonRpcControllerEthSpec
     response should haveResult(expectedBlockResponse)
   }
 
+  it should "handle eth_getBlockByHash request (block with checkpoint)" in new JsonRpcControllerFixture {
+    val blockToRequest = blockWithCheckpoint
+    val blockTd = blockToRequest.header.difficulty
+
+    blockchain
+      .storeBlock(blockToRequest)
+      .and(blockchain.storeTotalDifficulty(blockToRequest.header.hash, blockTd))
+      .commit()
+
+    val request = newJsonRpcRequest(
+      "eth_getBlockByHash",
+      List(JString(s"0x${blockToRequest.header.hashAsHexString}"), JBool(false))
+    )
+    val response = jsonRpcController.handleRequest(request).futureValue
+
+    val expectedBlockResponse =
+      Extraction.decompose(BlockResponse(blockToRequest, fullTxs = false, totalDifficulty = Some(blockTd)))
+
+    response should haveResult(expectedBlockResponse)
+  }
+
+  it should "handle eth_getBlockByHash request (block with treasuryOut)" in new JsonRpcControllerFixture {
```

Comment:
I'd prefer to be consistent with our naming so `treasuryOptOut` (besides `treasuryOut` sounds weird). This related to several val names and string occurrences.